src/app/test_executive/gossip_consistency.ml

```diff
@@ -0,0 +1,178 @@
+    let%bind receiver_pub_key = Util.pub_key_of_node receiver_bp in
+    let sender_bp =
+      Core_kernel.List.nth_exn (Network.block_producers network) 1
+    in
+    let%bind sender_pub_key = Util.pub_key_of_node sender_bp in
+    let num_payments = 3 in
+    let fee = Currency.Fee.of_int 10_000_000 in
+    let amount = Currency.Amount.of_int 10_000_000 in
+    [%log info] "gossip_consistency test: will now send %d payments"
+      num_payments ;
+    let%bind () =
+      send_payments ~logger ~sender_pub_key ~receiver_pub_key ~node:sender_bp
+        ~fee ~amount num_payments
+    in
+    [%log info]
+      "gossip_consistency test: sending payments done. will now wait for \
+       payments" ;
+    let%bind () =
+      wait_for_payments ~logger ~dsl:t ~sender_pub_key ~receiver_pub_key
+        ~amount num_payments
+    in
+    [%log info] "gossip_consistency test: finished waiting for payments" ;
+    let gossip_states = (network_state t).gossip_received in
+    let num_transactions_seen =
+      let open Gossip_state in
+      let ss =
+        Map.data gossip_states
+        |> List.map ~f:(Fn.compose By_direction.received transactions)
+      in
+      Set.(size (union ss))
+    in
+    [%log info] "gossip_consistency test: num_transactions_seen = %d"
+      num_transactions_seen ;
+    let%bind () =
+      if num_transactions_seen < num_payments - 1 then (
+        let result =
+          Malleable_error.soft_error_string ~value:()
+            (Printf.sprintf
+               "transactions seen = %d, which is less than (numpayments = %d) \
+                - 1"
+               num_transactions_seen num_payments)
+        in
+        [%log info]
+          "gossip_consistency test: TEST FAILURE.  transactions seen = %d, \
+           which is less than (numpayments = %d) - 1"
+          num_transactions_seen num_payments ;
+        result )
+      else
+        let result = Malleable_error.ok_unit in
+        [%log info] "gossip_consistency test: num_transactions_seen OK" ;
+        result
+    in
+    let `Seen_by_all inter, `Seen_by_some union =
+      Gossip_state.stats Transactions_gossip
+        (Map.data (network_state t).gossip_received)
+        ~exclusion_list:[Network.Node.id sender_bp]
+    in
+    [%log info] "gossip_consistency test: inter = %d; union = %d " inter union ;
+    let ratio =
+      if union = 0 then 1. else Float.of_int inter /. Float.of_int union
+      (* Gossip_state.consistency_ratio Transactions_gossip
+        (Map.data (network_state t).gossip_received) *)
+    in
+    [%log info] "gossip_consistency test: consistency ratio = %f" ratio ;
+    let threshold = 0.95 in
+    let%map () =
+      if ratio < threshold then (
+        let result =
+          Malleable_error.soft_error_string ~value:()
+            (Printf.sprintf
+               "consistency ratio = %f, which is less than threshold = %f"
+               ratio threshold)
+        in
+        [%log info]
```

Comment:
should this be `error` instead of `info`?

Reply:
i suppose that would be appropriate yes